xarray/core/variable.py

```diff
@@ -766,6 +769,7 @@ def chunk(self, chunks=None, name=None, lock=False):
             if utils.is_dict_like(chunks):
                 chunks = tuple(chunks.get(n, s)
                                for n, s in enumerate(self.shape))
+            data = indexing.WrapImplicitIndexing(data, indexing.OuterIndexer)
```

Comment:
Why is this line necessary here? (and only OuterIndexer?)
I think it would be better if we could have a simple comment above.

Reply:
Added a comment to explain, with a link to the upstream dask issue (https://github.com/dask/dask/issues/2883).